server/main.lua

```diff
@@ -434,55 +464,58 @@ end)
 lib.addCommand('transfervehicle', {help = locale('general.command_transfervehicle'), params = {
     {name = 'id', type = 'playerId', help = locale('general.command_transfervehicle_help')},
     {name = 'amount', type = 'number', help = locale('general.command_transfervehicle_amount'), optional = true}}}, function(source, args)
+    local qbx_core = exports.qbx_core
+    local qbx_vehicles = exports.qbx_vehicles
     local src = source
     local buyerId = args.id
     local sellAmount = args.amount or 0
+
     if src == buyerId then
-        return exports.qbx_core:Notify(src, locale('error.selftransfer'), 'error')
+        return qbx_core:Notify(src, locale('error.selftransfer'), 'error')
     end
     if saleTimeout[src] then
-        return exports.qbx_core:Notify(src, locale('error.sale_timeout'), 'error')
+        return qbx_core:Notify(src, locale('error.sale_timeout'), 'error')
     end
     if buyerId == 0 then
-        return exports.qbx_core:Notify(src, locale('error.Invalid_ID'), 'error')
+        return qbx_core:Notify(src, locale('error.Invalid_ID'), 'error')
     end
     local ped = GetPlayerPed(src)
 
     local targetPed = GetPlayerPed(buyerId)
     if targetPed == 0 then
-        return exports.qbx_core:Notify(src, locale('error.buyerinfo'), 'error')
+        return qbx_core:Notify(src, locale('error.buyerinfo'), 'error')
     end
 
     local vehicle = GetVehiclePedIsIn(ped, false)
     if vehicle == 0 then
-        return exports.qbx_core:Notify(src, locale('error.notinveh'), 'error')
+        return qbx_core:Notify(src, locale('error.notinveh'), 'error')
     end
 
-    local plate = qbx.string.trim(GetVehicleNumberPlateText(vehicle))
-    if not plate then
-        return exports.qbx_core:Notify(src, locale('error.vehinfo'), 'error')
-    end
+    local vehicleId = Entity(vehicle).state.vehicleid or qbx_vehicles:GetVehicleIdByPlate(GetVehicleNumberPlateText(vehicle))
+    if not vehicleId then return end
```

Comment:
This will occur if the vehicle is unowned, maybe worth adding a notification that the vehicle is not owned?